spark/src/main/scala/ai/chronon/spark/TableUtils.scala

```diff
@@ -134,67 +294,46 @@ case class TableUtils(sparkSession: SparkSession) {
     }
   }
 
+  def tableReadFormat(tableName: String): Format = tableFormatProvider.readFormat(tableName)
+
   // return all specified partition columns in a table in format of Map[partitionName, PartitionValue]
   def allPartitions(tableName: String, partitionColumnsFilter: Seq[String] = Seq.empty): Seq[Map[String, String]] = {
     if (!tableExists(tableName)) return Seq.empty[Map[String, String]]
-    if (isIcebergTable(tableName)) {
-      throw new NotImplementedError(
-        "Multi-partitions retrieval is not supported on Iceberg tables yet." +
-          "For single partition retrieval, please use 'partition' method.")
-    }
-    sparkSession.sqlContext
-      .sql(s"SHOW PARTITIONS $tableName")
-      .collect()
-      .map { row =>
-        {
-          val partitionMap = parsePartition(row.getString(0))
-          if (partitionColumnsFilter.isEmpty) {
-            partitionMap
-          } else {
-            partitionMap.filterKeys(key => partitionColumnsFilter.contains(key)).toMap
-          }
-        }
+    val format = tableReadFormat(tableName)
+    val partitionSeq = format.partitions(tableName)(sparkSession)
+    if (partitionColumnsFilter.isEmpty) {
+      partitionSeq
+    } else {
+      partitionSeq.map { partitionMap =>
+        partitionMap.filterKeys(key => partitionColumnsFilter.contains(key)).toMap
       }
+    }
   }
 
   def partitions(tableName: String, subPartitionsFilter: Map[String, String] = Map.empty): Seq[String] = {
     if (!tableExists(tableName)) return Seq.empty[String]
-    if (isIcebergTable(tableName)) {
+    val format = tableReadFormat(tableName)
+
+    if (format == Iceberg) {
```

Comment:
Consider having a supportSubPartitionsFilter method in Format. Stripe has another format that we would have (temp views) that also would not support subPartitionFilters.